src/firebase/auth/signUp.ts

```diff
@@ -0,0 +1,24 @@
+import firebase_app from '@/firebase/config';
+import {
+    getAuth,
+    createUserWithEmailAndPassword,
+    UserCredential,
+} from 'firebase/auth';
+
+const auth = getAuth(firebase_app);
+
+export default async function signUp(
+    email: string,
+    password: string
+): Promise<{ result: UserCredential | null; error: Error | null }> {
```

Comment:
you'll probably encounter this error after adding "use server" because you'll now be calling a server action when you call "signUp" in the signup pages, which are client components -- basically there are certain restrictions in place when you pass data between server and client components "Error: Only plain objects, and a few built-ins, can be passed to Client Components from Server Components. Classes or null prototypes are not supported."

so instead of returning the UserCredential object, return a serialized version of the result.